compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -644,6 +655,25 @@ private boolean isErrorTypeAssignable(BErrorType source, BErrorType target, Set<
                 isAssignable(source.detailType, target.detailType, unresolvedTypes);
     }
 
+    //TODO: recheck this.
+    private boolean isXMLTypeAssignable(BType sourceType, BType targetType, Set<TypePair> unresolvedTypes) {
+        int sourceTag = sourceType.tag;
+        int targetTag = targetType.tag;
+
+        if (targetTag == TypeTags.XML) {
```

Comment:
Since this method is called only after verifying that the source and target tags are XML, can't we change the method signature to accept XML types? This would get rid of redundant type tag checks such as this one

Reply:
The `sourceType` and `targetType` can be either `BXMLType` or `BXMLSubtype`. The way we have implemented. `BXMLSubtype` does not extend `BXMLType`. Hence, there isn't a common class we can use as parameter.